packages/runtime/src/next/preview-mode.tsx

```diff
@@ -54,6 +54,16 @@ if (originalUrl.searchParams.has(searchParamName)) {
 
   window.history.replaceState(null, '', url)
 }
+
+// Makeswift host connection check
+if (window.parent !== window) {
+  setInterval(() => {
+    window.parent.postMessage({ 
+      type: 'makeswift_connection_check',
+      payload: { currentUrl: window.location.href } 
+    }, appOrigin)
+  }, 20)
+}
```

Comment:
I feel like this should be part of a separate script, not the Preview Mode script, since it's not related to Preview Mode. Also, ideally we only send the heartbeat if we've established a connection instead of just sending a message every 20ms

Reply:
Yeah, that makes sense. I think I would like to keep the scripts separate internally for the purposes of code organization and then we can have a single script the API consumers use. That being said, it's ok if you want to leave it in there for now. We can come back later and split it up.

As for how we initialize, we can do something similar to Preview Mode, yes. You could even put the heartbeat inside the Preview Mode block if you want so we don't need two separate messages from the host (or we could have two listeners for that message)